### PR TITLE
feat: call hooks before and after parse

### PR DESCRIPTION
**Pull request by farnabaz**
### 🔗 Linked issue
#1119

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
Introduce two hooks to manipulate contents before/after parsing.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by Atinux — APPROVED** [review]

**Comment by Atinux** [comment]
Would be nice to add a test case and document it somewhere 😊

**Comment by Tahul** [comment]
This looks good to me, will it unlock proper reloading with Tailwind classes used in `.md` files?

I would only add a link to https://nitro.unjs.io/guide/plugins.html somewhere in the docs and specify where these plugins should be placed. 🙂

File: docs/content/4.api/4.advanced.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+title: 'Advanced'
+description: 'You can configure Nuxt Content with the content property in your nuxt.config.js|ts file.'
```
Comment (Atinux) [inline docs/content/4.api/4.advanced.md:3]:
```suggestion
description: 'Nuxt Content is highly customizable, giving you freedom and control over how the data is transformed.'
```

File: docs/content/4.api/4.advanced.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+Arguments:
+- file
+  - Type: `Object`
+  - Properties:
+    - _id: `String`
+    - body: `String`
+
+### Example
+
+Changing all appearances of react to vue in all Markdown files:
```
Comment (Atinux) [inline docs/content/4.api/4.advanced.md:28]:
```suggestion
Changing all occurrences of react to vue in all Markdown files:
```

File: docs/content/4.api/4.advanced.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+
+```ts [my-nitro-plugin.ts]
```
Comment (Atinux) [inline docs/content/4.api/4.advanced.md:31]:
```suggestion
```ts [server/plugins/content.ts]
```

File: docs/content/4.api/4.advanced.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+### `content:file:afterParse`
+
+Allows you to modify a document after being parsed by parsers.
+
+### Example
+
+Using content's first picture as cover image.
+
+```ts [my-nitro-plugin.ts]
```
Comment (Atinux) [inline docs/content/4.api/4.advanced.md:49]:
```suggestion
```ts [server/plugins/content.ts]
```

File: docs/content/4.api/4.advanced.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+---
+
+## Hooks
+
+The module adds some hooks you can use:
+
+::alert
+In order to use `content:file:*` hooks you need to create custom [nitro plugin](https://nitro.unjs.io/guide/plugins.html).
+::
+
```
Comment (Atinux) [inline docs/content/4.api/4.advanced.md:14]: Would be nice to add a section like:

```ts
export default defineNuxtConfig({
  // ...
  nitro: {
    plugins: ['~/server/content.ts']
  }
})
```

Hopefully we won't have to do it once https://github.com/nuxt/framework/issues/5211 will be available.

File: docs/content/4.api/4.advanced.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+icon: heroicons-outline:lightning-bolt
+---
+
+## Hooks
+
+The module adds some hooks you can use:
+
+`content:file:*` hooks are available in nitro runtime, in order to use them you need to create custom [nitro plugin](https://nitro.unjs.io/guide/plugins.html).
+
+- Create your custom plugin
```
Comment (Atinux) [inline docs/content/4.api/4.advanced.md:13]:
```suggestion
- Create a plugin in the `server/plugins/` directory
```

File: docs/content/4.api/4.advanced.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+  ```ts [server/plugins/content.ts]
+  export default defineNitroPlugin((nitroApp) => {
+    // ...
+  })
+  ```
+
+- Register your plugin in `nuxt.config.ts`
```
Comment (Atinux) [inline docs/content/4.api/4.advanced.md:21]:
```suggestion
- Register the plugin in `nuxt.config.ts`
```

**Pull request merged**